include/yaramod/types/expression.h

```diff
@@ -71,19 +71,23 @@ class Expression
 			default: return "Error - unknown type";
 		}
 	}
-	TokenStream* getTokenStream()
+	TokenStream* getTokenStream() const
 	{
 		return _tokenStream.get();
 	}
-	void setTokenStream(const std::shared_ptr<TokenStream>& ts)
+	const std::shared_ptr<TokenStream> getTokenStreamSharedPtr() const
```

Comment:
You are returning a copy here. You most probably wanted to return the reference.

Reply:
Thank you, I missed the & here.